src/grid/Grid.js

```diff
@@ -15,35 +13,38 @@ class Grid extends Component {
     activeOpacity: 1,
   }
 
+  styles = StyleSheet.create({
+    container: {
+      flex: 1,
+      flexDirection: this.isRow() ? 'column' : 'row',
+    },
+    opacityContainer: {
+      flex: 1,
+    },
+  })
+
   isRow() {
     let isRow = false;
     React.Children.forEach(this.props.children, (child) => {
-      if (child.type === Row) {
+      if (child && child.type === Row) {
         isRow = true;
       }
     });
 
     return isRow;
   }
 
-  componentWillMount() {
-    const {style} = this.props;
-
-    this.styles = {
-      flex: 1,
-      flexDirection: this.isRow() ? 'column' : 'row',
-      ...style,
-    };
-  }
-
   render() {
-    const {onPress, activeOpacity} = this.props;
+    const {onPress, activeOpacity, containerStyle} = this.props;
 
     if (onPress) {
       return (
-        <TouchableOpacity style={{flex: 1}} activeOpacity={activeOpacity} onPress={onPress}>
+        <TouchableOpacity style={opacityContainer} activeOpacity={activeOpacity} onPress={onPress}>
           <View
-            {...this.styles}
+            style={[
+              this.styles.container,
```

Comment:
Also once you move `styles` outside the class, this should be refactored as `styles.container`